src/amo/components/AddonReviewCard/index.js

```diff
@@ -294,33 +291,33 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
     const noAuthor = shortByLine || this.isReply();
 
     if (review) {
-      const timestamp = `
-        <a href="/${lang}/${clientApp}/addon/${
-        review.reviewAddon.slug
-      }/reviews/${review.id}/">
-          ${i18n.moment(review.created).fromNow()}
-        </a>
-      `;
+      const url = `/addon/${review.reviewAddon.slug}/reviews/${review.id}/`;
+      const timestamp = (
+        <Link to={url}>{i18n.moment(review.created).fromNow()}</Link>
+      );
+
       const byLineString = noAuthor
         ? // translators: Example in English: "posted last week"
           i18n.gettext('posted %(timestamp)s')
         : // translators: Example in English: "by UserName123, last week"
           i18n.gettext('by %(authorName)s, %(timestamp)s');
 
+      const localized = i18n.sprintf(byLineString, {
+        authorName: review.userName,
+        timestamp: '__timestamp__',
+      });
+
+      const parts = localized.split('__timestamp__');
+      const allParts = [parts.shift(), timestamp, ...parts];
```

Comment:
Rendering the remaining `...parts` isn't covered by any tests. It would be helpful to add coverage for it. You could do `sinon.stub(i18n, 'gettext')` to inject any kind of L10n string.